src/bin/pgcopydb/string_utils.c

```diff
@@ -683,10 +683,34 @@ pretty_print_bytes(char *buffer, size_t size, uint64_t bytes)
 	sformat(buffer, size, "%d %s", (int) count, suffixes[sIndex]);
 }
 
+/*
+ * pretty_print_bytes_per_second pretty prints bytes transmitted per second in
+ * a human readable form. Given 17179869184 it places the string
+ * "16 GBPS" in the given buffer.
+ */
+void
+pretty_print_bytes_per_second(char *buffer, size_t size, uint64_t bytes,
+							  uint64_t durationMs)
+{
+	/* avoid division by zero */
+	if (durationMs == 0)
+	{
+		sformat(buffer, size, "0 BPS");
+	}
+	else
+	{
+		char bytesPretty[BUFSIZE] = { 0 };
+		uint64_t bytesPerSecond = bytes * 1000 / durationMs;
+
+		pretty_print_bytes(bytesPretty, size-2, bytesPerSecond);
+		sformat(buffer, size, "%sPS", bytesPretty);
+	}
+}
+
```

Comment:
Please make sure to use official suffix for the unit, as in https://en.wikipedia.org/wiki/Data-rate_units in this case. Which I failed doing in `pretty_print_bytes` now that I read https://en.wikipedia.org/wiki/Byte again ; so that'd be for another patch then.

Reply:
I read through the wiki pages and come up with a decision. I believe it is best to mimic what `pg_size_pretty` function in pgcopydb. They use powers of 2, but do not use EIC standard. That means 1024 bytes = 1 kilobytes (kB). I like to keep a similar format and thus keep `pretty_print_bytes` as it is.

However, I failed to find anywhere in the PG codebase where a Data-rate unit is calculated and printed. I think that it makes more sense to use powers of 2. Do you think it makes sense to use EIC standard for Data-rate units? I mean 1024 bit/s (bit per second) = 1 Kibit/s (Kibibit per second)?

IMO, the problem is with how PG pretty prints sizes. They used a questionable format that causes more ambiguity and confusion when we start to report transmit rates. I think it has more significance to have a consistent format than the correct one. PG already messed up, and we need to keep things consistent with their format.

Reply:
Ok, I agree with you on using the Postgres way for pg_size_pretty and pretty_print_bytes, because internally we use both in different places and being able to compare the sizes is good.

For the rates, we can decide whether we want to use multiples of 1000 or 1024 (powers of 2 or powers of 10), but then I will insist on using the proper suffix for it. I am voting for using `kbit/s`, `Mbit/s`, and al, which means powers of 10.